### ISSUE TITLE
Version 1.3.0 - v8n().optional() and v8n().string() bug

### ISSUE THREAD

**Issue by ShaharMynd** [imbrn/v8n#149]
Hey, when I updated to version 1.3.0

One of my function stopped working, I managed to see that the v8n.optional() with v8n.string() is causing the bug,

# Reproducing
```
var a = v8n().schema({name: v8n().optional(v8n().string())}).testAll({name: ""})
```

In version 1.2.3 there is no error:
![image](https://user-images.githubusercontent.com/42337023/58021122-c4080b00-7b12-11e9-9f11-99e2955202b6.png)

But in version 1.3.0
```
 var a = v8n().schema({name: v8n().optional(v8n().string())}).testAll({name: ""})
undefined
> a
[ { Error
      at /Users/shaharshalev/Documents/repos/webapp-backend/node_modules/v8n/dist/v8n.cjs.js:224:20
      at Array.forEach (<anonymous>)
      at Proxy.testAll (/Users/shaharshalev/Documents/repos/webapp-backend/node_modules/v8n/dist/v8n.cjs.js:220:18)
      at repl:1:62
      at Script.runInThisContext (vm.js:119:20)
      at REPLServer.defaultEval (repl.js:332:29)
      at bound (domain.js:395:14)
      at REPLServer.runBound [as eval] (domain.js:408:12)
      at REPLServer.onLine (repl.js:639:10)
      at REPLServer.emit (events.js:194:15)
    rule:
     Rule { name: 'schema', fn: [Function], args: [Array], modifiers: [] },
    value: { name: '' },
    cause: [ [Error] ],
    target: undefined } ]
```

**Comment by imbrn** [comment]
Thank you @ShaharMynd. I'm looking at it.

**Comment by imbrn** [comment]
Fixed by #150
